src/session_state.rs

```diff
@@ -19,12 +19,13 @@ pub struct SessionState<C: Clock> {
     pub client_id: String<64>,
     last_transmission: Option<Instant<C>>,
     pub pending_subscriptions: Vec<u16, 32>,
+    pub pending_publish: LinearMap<u16, (usize, [u8; T]), 16>,
```

Comment:
We'll definitely want to make this generic to allow people to tune the RAM requirements of the library to their application. Alternatively, we could restrict this to 1 or 2 messages.

Reply:
Also, we should make this a map of `Vec<u8, T>` to make it easy to copy messages around without having to maintain an array + len

Reply:
I don't know the preferred style here (const generics aren't in the Book so I'm not even sure where to look for it), as it's a generic I would assume short names, but `pub struct SessionState<C: Clock, const T: usize, const U: usize>` hardly seems readable, especially with prospects of adding more later.

Would `pub struct SessionState<C: Clock, const MaxMessageSize: usize, const QoS1Buffe: usizer>` be too much?

Reply:
Naming preference from my end would be all caps, underscored spaces. It also helps readability o put the trait bounds in a `where` clause.

E.g.
```rust
pub struct SessionState<CLOCK, const MSG_SIZE, const NUM_MESSAGES>
where
    CLOCK: Clock,
    MSG_SIZE: usize,
    NUM_MESSAGES: usize
{
}
```

Reply:
It turns out const generics can't be really used in where clauses (or it's not documented and I didn't happen to ned up with the right permutation)

Also rustc complains about the casing - weirdly enough it complains about `TCP_STACK` but not `CLOCK`. So I went with what rustc wanted and changed `CLOCK` to `Clock` for consistency.

warning: type parameter `TCP_STACK` should have an upper camel case name
  --> src\mqtt_client.rs:98:19
   |
98 | pub struct Minimq<TCP_STACK, CLOCK, const MSG_SIZE: usize, const MSG_COUNT: usize>
   |                   ^^^^^^^^^ help: convert the identifier to upper camel case: `TcpStack`
   |
   = note: `#[warn(non_camel_case_types)]` on by default

Reply:
Ah, I didn't realize that rustfmt prefers camel case for type parameters. If that's the case, go with whatever rustfmt says is correct. I prefer to rely on the automated tools to keep things consistent over my own personal preference. Thanks for pointing this out to me!